src/main/java/graphql/ExecutionInput.java

```diff
@@ -245,27 +268,87 @@ public Builder localContext(Object localContext) {
         }
 
         /**
-         * By default you will get a {@link GraphQLContext} object but you can set your own.
+         * The legacy context object
          *
          * @param context the context object to use
+         *
          * @return this builder
+         *
+         * @deprecated - use {@link #graphQLContext(GraphQLContext)} instead
          */
+        @Deprecated
         public Builder context(Object context) {
             this.context = context;
             return this;
         }
 
+        /**
+         * The legacy context object
+         *
+         * @param contextBuilder the context builder object to use
+         *
+         * @return this builder
+         *
+         * @deprecated - use {@link #graphQLContext(GraphQLContext.Builder)} instead
+         */
+        @Deprecated
         public Builder context(GraphQLContext.Builder contextBuilder) {
             this.context = contextBuilder.build();
             return this;
         }
 
+        /**
+         * The legacy context object
+         *
+         * @param contextBuilderFunction the context builder function to use
+         *
+         * @return this builder
+         *
+         * @deprecated - use {@link #graphQLContext(UnaryOperator)} instead
+         */
+        @Deprecated
         public Builder context(UnaryOperator<GraphQLContext.Builder> contextBuilderFunction) {
             GraphQLContext.Builder builder = GraphQLContext.newContext();
             builder = contextBuilderFunction.apply(builder);
             return context(builder.build());
         }
 
+        /**
+         * This allows you to set up your own context object that will be passed to all data fetchers during execution
+         *
+         * @param context the context object to use
+         *
+         * @return this builder
+         */
+        public Builder graphQLContext(GraphQLContext context) {
+            this.graphQLContext = assertNotNull(context, () -> "you MUST provide a non null GraphqlContext");
+            return this;
+        }
+
+        /**
+         * This allows you to set up your own context object that will be passed to all data fetchers during execution
+         *
+         * @param contextBuilder the {@link GraphQLContext.Builder} object to use
+         *
+         * @return this builder
+         */
+        public Builder graphQLContext(GraphQLContext.Builder contextBuilder) {
+            return graphQLContext(contextBuilder.build());
+        }
+
```

Comment:
I'm wondering if there is even a need to replace the `GraphQLContext` instance? A single instance, not exposed for replacement, seems fine for a shared context, or at least it could start that way until there is evidence otherwise. I can't think of reasons to replace it and if they do surface, it could rather lead to enhancing`GraphQLContext` itself.

Reply:
This is about trying to seed it up with information from outside the graphql query run

Eg imagine some how you now the principal of the user making the request at the time - you can set this in via this.

Of course you could do

```
ExecutionInput ei - ExecutionInput.newExecutionInput().query(s).variables(..).build()

ei.getGraphqlContext().put(Principal.class, requestPrincipal)
```

Reply:
I can see how one can insert key-value pairs into the context. It could be done through:
- `ExecutionInput` instance via `getGraphQLContext`
- `ExecutionInput.Builder` instance via `graphQLContext(UnaryOperator<GraphQLContext.Builder>)` although this one currently replaces the underlying context instance rather than transforming it (as per my next comment) so doesn't quite work for just seeding (without replacing) the context.

What I don't see is why even allow replacing the `GraphQLContext` instance? If all you want to seed it, that should be possible with a single, shared instance only that isn't exposed to be replaced.

Reply:
gerat idea - done - goes into the one and only context